lib/jar.ts

```diff
@@ -28,47 +28,90 @@ interface MavenPackageInfo {
   v: string; // version
 }
 
-export async function getTempPomPathForJar(
-  root: string,
-  jarName: string,
-): Promise<string> {
-  const jarPath = path.resolve(root, jarName);
+function getSha1(buf: Buffer) {
+  return crypto
+    .createHash(ALGORITHM)
+    .update(buf)
+    .digest(DIGEST);
+}
 
+async function getMavenDependency(
+  targetPath: string,
+): Promise<MavenDependency> {
   try {
-    if (!fs.existsSync(jarPath)) {
-      throw new Error('Unable to find jar at ' + jarPath);
-    }
+    const contents = fs.readFileSync(targetPath);
+    const sha1 = getSha1(contents);
+    const { g, a, v } = await getMavenPackageInfo(sha1);
+    return {
+      groupId: g,
+      artifactId: a,
+      version: v,
+    };
+  } catch (err) {
+    throw new Error('Failed to get maven dependency. ' + err.message);
+  }
+}
 
-    const fileContents = fs.readFileSync(jarPath);
+async function getDependencies(paths: string[]): Promise<MavenDependency[]> {
+  let dependencies: MavenDependency[] = [];
+  for (const p of paths) {
+    const dependency = await getMavenDependency(p);
```

Comment:
Put a try catch around this and debug when we fail to get a dependency (some will be on maven central, while others wont).

Reply:
we have catch on getMavenDependency, is that ok?

Reply:
but you throw from getMavenDependency. I think we should continue processing even if we don't have info for all jarsin the folder

Reply:
```suggestion
    try {
      const dependency = await getMavenDependency(p);
      dependencies.push(dependency);
    } catch (err) {
      debug(err);
    }
```

Reply:
```suggestion
    try {
      const dependency = await getMavenDependency(p);
      dependencies.push(dependency);
    } catch (err) {
      debug(err);
    }
  }
  if (dependencies.length === 0) { // no dependencies at all found -> throw }
```

Reply:
ok, pushed update to log errors and continue and added test for that